lib/bloc/pos_catalog/sqlite/repository.dart

```diff
@@ -198,4 +198,48 @@ class SqliteRepository implements Repository {
       ),
     );
   }
+
+  @override
+  Future<PosReportResult> fetchSalesReport(DateTime start, DateTime end) async {
+    final db = await getDB();
+    final report = await db.rawQuery(
+      """
+      SELECT
```

Comment:
@ademar111190 I think that the requirement is:
1. for sats - to show always the sum of all sales
2. for fiat - to show sum of fiat only if ALL sale lines in the period has the same currency.

So I think this sql should do it:

```
SELECT
        sum((quantity * price_per_item)) as fiat_value,
        sum((quantity * price_per_item * sat_conversion_rate)) as sat_value,
        count(distinct(sale_id)) as total_sales,
        currency
      FROM sale_line
      LEFT JOIN sale ON sale_line.sale_id = sale.id
      WHERE (sale.date >= ? and sale.date <= ?)
      GROUP BY currency
      limit 2;
```
      
Note that it is based on your query with the following changes:
1. without grouping by sale_id (why did you need that?) 
2. added count for sales
3. added limit to optimize a bit.

If the result set contains only one line then we should show the fiat value, if we get more than one line than we know there are multiple currencies in use and we show only sats in the report.
Does it make sense?

Reply:
It seems to be working 😄